orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/PublishKeelSpecsTask.kt

```diff
@@ -0,0 +1,131 @@
+import com.netflix.spinnaker.orca.keel.model.ResourceSpec
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import retrofit.client.Response
+import java.util.concurrent.TimeUnit
+
+@Component
+class PublishKeelSpecsTask
+@Autowired
+constructor(
+  private val keelService: KeelService,
+  private val keelObjectMapper: ObjectMapper,
+  private val scmService: ScmService
+) : RetryableTask {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun execute(stage: Stage): TaskResult {
+//      parameters:
+//      - name: repoUrl
+//        mapping: cluster.env.MD_REPO_URL
+//        label: Git Repository SSH URL
+//        order: 0
+//      - name: directory
+//        mapping: cluster.env.MD_DIRECTORY
+//        label: File Directory
+//        defaultValue: .netflix/
+//        order: 1
+//      - name: fileExtension
+//        mapping: cluster.env.MD_FILE_EXTENSION
+//        label: File Extension
+//        defaultValue: .yml
+//        order: 2
+//      - name: forceRepublish
+//        mapping: cluster.env.MD_FORCE_REPUBLISH
+//        defaultValue: false
+//        label: Force Republish
+//        order: 3
+//      - name: gitBranch
+//        mapping: cluster.env.GIT_BRANCH
+//        defaultValue: ${trigger.branch}
+//        label: Branch
+//        order: 4
+//      - name: gitSha
+//        mapping: cluster.env.GIT_SHA
+//        defaultValue: ${trigger.hash}
+//        label: SHA
+//        order: 5
+
+    val context = keelObjectMapper.convertValue<PublishKeelSpecsContext>(stage.context)
+    val manifests = scmService.listKeelManifests(
+      context.scmType, context.projectKey, context.repositorySlug, context.ref)
+
+    var failedManifest: String? = null
+    for (manifest in manifests) {
+      // TODO: do we even support multiple SCM servers of each kind? judging from igor, I'm guessing no
+      val uri = "${context.scmType}://<server???>/${context.projectKey}/${context.repositorySlug}/.netflix/spinnaker/$manifest"
+
+      log.info("Retrieving keel manifest at $uri")
+      // FIXME: I *think* the ref parameter below should actually be based on info from the (Rocket?) code event trigger
+      val manifestContents = scmService.getKeelManifest(
+        context.scmType, context.projectKey, context.repositorySlug, manifest, context.ref)
+
+      var resourceSpec: ResourceSpec
+      try {
+        resourceSpec = keelObjectMapper.convertValue(manifestContents)
+      } catch (e: java.lang.Exception) {
+        throw IllegalArgumentException("Unable to parse resource spec at $uri")
+      }
+
+      // TODO: confirm this is the right thing to do
+      val user = stage.execution.trigger.user ?: "anonymous"
+      log.info("Publishing manifest $manifest to keel on behalf of $user")
+      val response = keelService.publishSpec(resourceSpec, user)
```

Comment:
let's assume for now that the only flow is this:
- you have one file
- it must be called `spinnaker.yml`
- it it is published to the delivery config endpoint, not the resource endpoint

This is a change based on https://github.com/spinnaker/keel/issues/660